Add support for `write.metadata.path` (#1642)
apache/iceberg-python: mkdocs/docs/configuration.md

## Location Providers

Apache Iceberg uses the concept of a `LocationProvider` to manage file paths for a table's data and metadata files. In PyIceberg, the
`LocationProvider` module is designed to be pluggable, allowing customization for specific use cases. The
`LocationProvider` for a table can be specified through table properties.

While data files can leverage provider-specific optimizations, metadata files always follow a simple path configuration. Regardless of

> the `LocationProvider` used, metadata files are written to the path specified by the [`write.metadata.path` table configuration](#write-options) table property.

Reviewer: As it stands, I think a custom `LocationProvider` can reasonably override (*re-declare* really, because it's Python) the metadata-related methods. Maybe relates to [this discussion](https://github.com/apache/iceberg-python/pull/1642/files#r1954666388)

Reviewer 2: ```suggestion
Both data file and metadata file locations can be customized by configuring the table properties [write.data.path and write.metadata.path](#write-options), respectively.

For more granular control, you can override the `LocationProvider`'s `new_data_location` and `new_metadata_location` methods to define custom logic for generating file paths.
```

## Location Providers

> Apache Iceberg uses the concept of a `LocationProvider` to manage file paths for a table's data and metadata files. In PyIceberg, the
> `LocationProvider` module is designed to be pluggable, allowing customization for specific use cases. The
> `LocationProvider` for a table can be specified through table properties.

Reviewer: Following the other discussion I think "Apache Iceberg ... managing metadata files" is a bit misleading given it's just PyIceberg. How about:

```suggestion
Apache Iceberg uses the concept of a `LocationProvider` to manage file paths for a table's data files. In PyIceberg, the
`LocationProvider` module is designed to be pluggable, allowing customization for specific use cases, and to additionally determine metadata file locations. The
`LocationProvider` for a table can be specified through table properties.
```

> Both data file and metadata file locations can be customized by configuring the table properties [write.data.path and write.metadata.path](#write-options), respectively.

Reviewer 3: ```suggestion
Both data file and metadata file locations can be customized by configuring the table properties [`write.data.path` and `write.metadata.path`](#write-options), respectively.
```

> For more granular control, you can override the `LocationProvider`'s `new_data_location` and `new_metadata_location` methods to define custom logic for generating file paths.

Reviewer: Nit: This only applies to custom location providers. *We* understand that and that they can be loaded but I think a new reader doesn't have that context at this point. How about appending something like " - see loading custom location providers" and linking `#loading-a-custom-location-provider` there?

Reviewer 2: ```suggestion
For more granular control, you can override the `LocationProvider`'s `new_data_location` and `new_metadata_location` methods to define custom logic for generating file paths. See [`Loading a Custom Location Provider`](configuration.md#loading-a-custom-location-provider).
```
